src/guiguts/footnotes.py

```diff
@@ -122,28 +143,957 @@ def join_to_previous(self) -> None:
             f"{fn_cur_start} -1l linestart", f"{fn_cur_end} +1l linestart"
         )
         maintext().delete(f"{fn_prev_end} -1c", f"{fn_prev_end} lineend")
-        maintext().insert(fn_prev_end, "\n" + continuation_text + "\n")
+        maintext().insert(fn_prev_end, "\n" + continuation_text)
         self.checker_dialog.remove_entry_current()
-        maintext().see(fn_prev_end)
+        # Note index of the record to which we joined the continuation text.
+        saved_prev_rec_index = fn_index - 1
+        self.run_check()
+        display_footnote_entries(auto_select_line=False)
+        fn_records = self.get_fn_records()
+        prev_record = fn_records[saved_prev_rec_index]
+        # Get index into dialog entries for this record.
+        dialog_entry_index = self.map_fn_record_to_dialog_index(prev_record)
+        if dialog_entry_index < 0:
+            logger.error("Unexpected return from 'join_to_previous()'")
+            return
+        # Make it the selected entry in the dialog.
+        self.checker_dialog.select_entry_by_index(dialog_entry_index)
+
+    def autoset_chapter_lz(self) -> None:
+        """Insert a 'FOOTNOTES:' LZ header line before each chapter break.
+
+        Before inserting the first LZ header, skip the first 200 lines of the
+        file if it is longer than 200 lines. This avoids inserting LZ headers
+        at pseudo paragraph breaks in front matter pages.
```

Comment:
This is true of most projects with footnotes, but I feel pretty uncomfortable with this as a hard threshold (even if it's also true of GG1), without some way for those with projects where the footnotes start before that threshold to deal with it easily. One of the test projects I've downloaded has the first footnote anchor before line 200, even though the footnote itself is after; another has two footnotes (both anchor and footnote, before line 200). I browsed the PP pool, and found another that had first footnote and anchor preceding line 200.

Reply:
Since this is the same as GG1, I'd be happy for it to be merged with the 200 threshold. However, it occurs to me that we could maybe improve on GG1's algorithm by starting chapter LZs after the location of the first footnote, rather than after a fixed line number. As I said, happy for that to go into future footnote improvements, if not in this PR.

Reply:
This should be simple to fix. I will add to this PR the @windymilla suggestion of starting chapter LZs after the first footnote rather than after the first 200 lines. Thank you @srjfoo for highlighting the issue.